ddev/ddev
# fix: add CI to instrumentation tags

## The Issue

I read this page https://ddev.com/blog/stats-on-ddev-usage-nov-2024/ and found

> ... `ddev drush` is excluded here because it was used 515,000 times in the last 7 days. I guess some people must be using it in an automated situation. If you’re using any of these in CI, please turn off reporting if it happens to be on, `ddev config global --instrumentation-opt-in=false` or `export DDEV_NO_INSTRUMENTATION=true`.

~instead of asking user to set `DDEV_NO_INSTRUMENTATION` env var, let's make DDEV automatically detect and set no instrumentation if it is run on a CI pipeline~

here are some popular CI tools that have `CI=true` env var 

- Github Actions https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables
- Gitlab CI https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
- Buildkite https://buildkite.com/docs/pipelines/configure/environment-variables#CI
- Bitbucket https://support.atlassian.com/bitbucket-cloud/docs/variables-and-secrets/
- CircleCI https://circleci.com/docs/variables/#built-in-environment-variables



## How This PR Solves The Issue

## Manual Testing Instructions

* Build using the development AmplitudeAPIKey
* Do some things and then `ddev debug instrumentation flush`
* Verify that `CI` comes through as false in https://app.amplitude.com/analytics/ddev/live-events using "DDEV - Development"
* export CI=true
* Do over, verify CI=true

## Automated Testing Overview



## Release/Deployment Notes

- [ ] Followup in the future after this goes into release and has time for people to have it, we can update our reports to test whether CI is true.
